usefulscripts/dumpModules.js

```diff
@@ -0,0 +1,112 @@
+var modules ={
+		'usb' : '0100000000000006',
+		'tma' : '0100000000000007',
+		'boot2': '0100000000000008',
+		'settings' : '0100000000000009',
+		'bus' : '010000000000000A',
+		'bluetooth' : '010000000000000B',
+		'bcat' : '010000000000000C',
+		'friends': '010000000000000E',
+		'nifm': '010000000000000F',
+		'ptm': '0100000000000010',
+		'bsdsockets': '0100000000000012',
+		'hid': '0100000000000013',
+		'audio': '0100000000000014',
+		'LogManager.Prod' : '0100000000000015',
+		'wlan' : '0100000000000016',
+		'ldn' : '0100000000000018',
+		'nvservices' : '0100000000000019',
+		'pcv' : '010000000000001A',
+		'ppc' : '010000000000001B',
+		'nvnflinger' : '010000000000001C',
+		'pcie.withoutHb' : '010000000000001D',
+		'account' : '010000000000001E',
+		'ns' : '010000000000001F',
+		'nfc' : '0100000000000020',
+		'psc' : '0100000000000021',
+		'capsrv' : '0100000000000022',
+		'am' : '0100000000000023',
+		'ssl' : '0100000000000024', 
+		'nim' : '0100000000000025',
+		'lbl' : '0100000000000029',
+		'btm' : '010000000000002A',
+		'erpt' : '010000000000002B',
+		'vi' : '010000000000002D',
+		'pctl' : '010000000000002E',
+		'npns' : '010000000000002F',
+		'eupld': '0100000000000030',
+		'glue' : '0100000000000031',
+		'eclct' : '0100000000000032',
+		'es' : '0100000000000033',
+		'fatal' : '0100000000000034',
+		'creport' : '0100000000000036',
+		
+};
+function dumpModule(module, loader, name) {
+	//We need a ILocationResolver to pass to fsp to say what we are reading so we're getting a handle
+	sc.getService("lr", (lripc) => {
+		//3 is the StorageID for NAND System
+		var lr = sc.ipcMsg(0).data(3).sendTo(lripc);
+		sc.withHandle(lr.movedHandles[0], (content) => {
+		        //We are getting our ContentPath needed for fsp, c being the "receiving" buffer 
+			var buf = new ArrayBuffer(0x300);
+			sc.ipcMsg(0).data(utils.parseAddr(module)).cDescriptor(buf).sendTo(content);
+			
+			//We are now mounting our code region
+			var fs =sc.ipcMsg(0).datau64(utils.parseAddr(module)).xDescriptor(buf).sendTo(loader).assertOk();
+			sc.withHandle(fs.movedHandles[0], (storage) => {
+				//utils.log('Got IFileSystem handle: 0x'+ storage.toString(16));
+				var fs = new sc.IFileSystem(sc, storage);
+				var dir = fs.OpenDir('/').getValue();
+				//DUMP ALL THE THINGS
+				dir.DirDump(name);
+			});
+		});
+	});
+}
+
+utils.log("stage1, getting webkit ldr:ro handle");
+//We are reusing WebKit's ldr:ro session 
+var ldrro_mng_ptr = utils.add2(sc.mainaddr, 0x955558);
```

Comment:
why don't you open your own `ldr:ro` session? are they that limited or something?

Reply:
The bug doesn't work if you open your own